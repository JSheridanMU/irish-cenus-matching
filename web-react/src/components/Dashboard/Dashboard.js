import React from 'react'
import { useTheme } from '@material-ui/core/styles'
import { Grid, Paper } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import clsx from 'clsx'

import PersonCount from '../PersonCount/PersonCount'
import PersonData from '../PersonData/PersonData'
export default function Dashboard() {
  const theme = useTheme()

  const useStyles = makeStyles((theme) => ({
    root: {
      display: 'flex',
    },
    paper: {
      padding: theme.spacing(2),
      display: 'flex',
      overflow: 'auto',
      flexDirection: 'column',
    },
    fixedHeight: {
      height: 240,
    },
  }))
  const classes = useStyles(theme)
  const fixedHeightPaper = clsx(classes.paper, classes.fixedHeight)

  return (
    <React.Fragment>
      <Grid container spacing={4}>
        {/* Person Counts */}
        <Grid item xs={12} md={4} lg={3}>
          <Paper className={fixedHeightPaper}>
            <PersonCount year="1901" />
          </Paper>
        </Grid>
        <Grid item xs={12} md={4} lg={3}>
          <Paper className={fixedHeightPaper}>
            <PersonCount year="1911" />
          </Paper>
        </Grid>
        {/* Person Format */}
        <Grid item xs={12}>
          <Paper className={classes.paper}>
            <PersonData />
          </Paper>
        </Grid>
      </Grid>
    </React.Fragment>
  )
}
