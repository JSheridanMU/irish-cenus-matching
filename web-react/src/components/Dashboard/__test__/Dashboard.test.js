import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { ApolloProvider, ApolloClient, InMemoryCache } from '@apollo/client'
import Dashboard from '.././Dashboard'

afterEach(cleanup)

const client = new ApolloClient({
  uri: process.env.REACT_APP_GRAPHQL_URI,
  cache: new InMemoryCache(),
})

test('renders without crashing', () => {
  render(
    <ApolloProvider client={client}>
      <Dashboard />
    </ApolloProvider>
  )
})

test('renders with temporary content', () => {
  const { getByText, getAllByText } = render(
    <ApolloProvider client={client}>
      <Dashboard />
    </ApolloProvider>
  )

  getByText('1901')
  getByText('1901')
  getAllByText('Loading')
})

test('matches snapshot', () => {
  const tree = renderer
    .create(
      <ApolloProvider client={client}>
        <Dashboard />
      </ApolloProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
