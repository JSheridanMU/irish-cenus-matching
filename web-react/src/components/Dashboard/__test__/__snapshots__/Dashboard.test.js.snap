// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      className="MuiPaper-root makeStyles-paper-12 makeStyles-fixedHeight-13 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        1901
      </h2>
      <p
        className="MuiTypography-root MuiTypography-h4"
      >
        Loading...
      </p>
      <p
        className="MuiTypography-root makeStyles-depositContext-14 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        people found in
      </p>
      <p
        className="MuiTypography-root MuiTypography-h4"
      >
        Loading...
      </p>
      <p
        className="MuiTypography-root makeStyles-depositContext-14 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        households
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      className="MuiPaper-root makeStyles-paper-12 makeStyles-fixedHeight-13 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        1911
      </h2>
      <p
        className="MuiTypography-root MuiTypography-h4"
      >
        Loading...
      </p>
      <p
        className="MuiTypography-root makeStyles-depositContext-14 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        people found in
      </p>
      <p
        className="MuiTypography-root MuiTypography-h4"
      >
        Loading...
      </p>
      <p
        className="MuiTypography-root makeStyles-depositContext-14 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        households
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <div
      className="MuiPaper-root makeStyles-paper-12 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <p>
        Loading
      </p>
    </div>
  </div>
</div>
`;
